cegui/src/text/LegacyTextParser.cpp

```diff
@@ -163,9 +162,14 @@ bool LegacyTextParser::parse(const String& inText, std::u32string& outText,
 
                 if (escaped)
                 {
-                    // If it is not actually escaped, print the slash itself too.
+                    // We got \n, so make line break
+                    if (codePoint == 'n')
+                    {
+                        codePoint = '\n';
+                    }
```

Comment:
This is a wrong place and at the same time the logic is inconsistent.
First, now you may render the text without a parser, and with your approach escaped characters will suddenly stop working. Also, every new parser will have to handle escaped characters. More robust would be to use another tag like [br] for this, as in HTML, making this explicitly belonging to a certain parser. Or use real newlines in the text, they are parsed by the common text rendering logic which is independent from the parser used.
Second, when you support \n, you should immediately think about \t, \r etc and why '\n' is an escape sequence and '\a' is literally "\a". Again, using [br] will solve this dilemma out of the box.

BTW, LegacyTextParser is called so because I wanted to implement a new text parser based on well known HTML and CSS text formatting tags and with a stack to enable intuitive nesting. But I really have no time for it now so you may do it if you are interested in text formatting, and I can help with reviewing and polishing the code. Another nice feature in my plan was a recursive argument substitution. If you are interested, we can discuss requirements and API.

Reply:
\n was backward compatibility (it was supported by DefaultRenderedStringParser), but I can change to [br]

Reply:
I think about html too (or `[]` tags wit multiple parameters), but no time too ... This PR is only moved stuff from my old custom parser (instead of ported it to new API). Main goal was image scaling based on aspect ratio and fix color for black text. For very advanced formatting I use CEF (and rendering to texture).